Pass `--fix` option to `ruff` in pre-commit

## Problem

<!-- Describe a problem solved by this feature; or delete the section entirely. -->
`ruff` is able to automatically fix some things like unused imports, except that we aren't passing the `--fix` argument to enable that. eg:

```
catalog/dags/create_filtered_index/dag_factory.py:58:41: F401 [*] `airflow.models.baseoperator.chain` imported but unused
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

## Description

<!-- Describe the feature and how it solves the problem. -->
Update the `ruff` pre-commit hook to pass the `--fix` arg:

```diff
diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index 5e39c541d..4485487f3 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -62,6 +62,8 @@ repos:
     rev: "v0.0.261"
     hooks:
       - id: ruff
+        args:
+          - --fix
 
   - repo: https://github.com/ambv/black
     rev: 23.3.0
```

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -62,6 +62,8 @@ repos:
     rev: "v0.0.261"
     hooks:
       - id: ruff
+        args:
+          - --fix
 
   - repo: https://github.com/ambv/black
     rev: 23.3.0

